lua/apisix/plugins/wolf-rbac.lua

```diff
@@ -0,0 +1,380 @@
+        core.log.error("FAIL REQUEST [ ",core.json.delay_encode({method=method, uri=uri, body=body, headers=myheaders}), " ] failed! res is nil, err:", err)
+        return nil, err
+    end
+
+    return res
+end
+
+local function http_get(uri, myheaders, timeout)
+    return http_req("GET", uri, nil, myheaders, timeout)
+end
+
+local function http_post(uri, body, myheaders, timeout)
+    return http_req("POST", uri, body, myheaders, timeout)
+end
+
+local function http_put(uri,  body, myheaders, timeout)
+    return http_req("PUT", uri, body, myheaders, timeout)
+end
+
+function _M.check_schema(conf)
+    core.log.info("input conf: ", core.json.delay_encode(conf))
+
+    local ok, err = core.schema.check(schema, conf)
+    if not ok then
+        return false, err
+    end
+
+    if not conf.appid then
+        conf.appid = 'unset'
+    end
+    if not conf.server then
+        conf.server = 'http://127.0.0.1:10080'
+    end
+
+    return true
+end
+
+
+local function fetch_rbac_token()
+    local args = ngx.req.get_uri_args()
+    if args and args.rbac_token then
+        return args.rbac_token
+    end
+
+    local headers = ngx.req.get_headers()
+    if headers.Authorization then
+        return headers.Authorization
+    end
+    if headers['x-rbac-token'] then
+        return headers['x-rbac-token']
+    end
+    local cookie, err = ck:new()
+    if not cookie then
+        return nil, err
+    end
+    local val, err = cookie:get("x-rbac-token")
+    return val, err
+end
+
+local function loadjson(str)
+    local ok, jso = pcall(function() return cjson.decode(str) end)
+    if ok then
+        return jso
+    else
+        return nil, jso
+    end
+end
+
+local function check_url_permission(server, appid, action, resName, clientIP, wolf_token)
+    local retry_max = 3
+    local errmsg = nil;
+    local userInfo = nil
+    local res = nil
+    local err = nil
+    local access_check_url = server .. "/wolf/rbac/access_check"
+    local headers = new_headers()
+    headers["x-rbac-token"] = wolf_token
+    headers["Content-Type"] = "application/json; charset=utf-8"
+    local args = { appID = appid, resName = resName, action = action, clientIP=clientIP}
+    local url = access_check_url .. "?" .. ngx.encode_args(args)
+    local timeout = 1000 * 10
+
+    for i = 1, retry_max do
+        -- TODO: read apisix info.
+        res, err = http_get(url, headers, timeout)
+        if err then
+            errmsg = 'check permission failed!' .. tostring(err)
+            break
+        else
+            core.log.info("check permission request:", url, ", status:", res.status, ",body:", core.json.delay_encode(res.body))
+            if res.status < 500 then
+                break
+            else
+                core.log.info("request [curl -v ", url, "] failed! status:", res.status)
+                if i < retry_max then
+                    ngx.sleep(100)
+                end
+            end
+        end
+    end
+
+    if err then
```

Comment:
this is a bad code style, please take a look at this new style:

```lua
if err then
    -- ...
end

if res.status ~= 200 and res.status ~= 401 then
    return {status=500, err='request to wolf-server failed, status:' .. tostring(res.status)}
end

-- ... other code

```

Reply:
Done